### PR TITLE
Add `translate` method to `Interaction`

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

This PR, as [discussed](https://discord.com/channels/336642139381301249/812600992116899861/1010223728933752905) on the [Discord server](https://discord.com/invites/dpy), adds a method called `translate` to `Interaction` that makes it easier to translate command responses and more.

```py
@tree.command()
async def foo(interaction: Interaction):
    response = "Hello, this is a test command!"

    # Before
    from discord.app_commands import TranslationContext, TranslationContextLocation, locale_str

    translator = tree.translator
    if not translator:
        await interaction.response.send_message(response)
        return

    context = TranslationContext(
        TranslationContextLocation.other,
        data=interaction.command
    )
    translated_response = await translator.translate(
        string=locale_str(response),
        locale=interaction.locale,
        context=context
    )

    # After
    translated_response = await interaction.translate(response) or response
    # .translate returns None if no translator is set or there was no translation returned
    # `or` handles that.


    await interaction.response.send_message(translated_response)
```
## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/interactions.py [hunk 1]
```diff
@@ -503,6 +504,44 @@ async def delete_original_response(self) -> None:
             proxy_auth=http.proxy_auth,
         )
 
+    async def translate(
+        self, string: Union[str, locale_str], *, locale: Locale = MISSING, data: Optional[Any] = MISSING
+    ) -> Optional[Any]:
```
Comment (nerdguyahmad) [inline discord/interactions.py:509]:
```suggestion
    ) -> Optional[str]:
```

File: discord/interactions.py [hunk 2]
```diff
@@ -503,6 +504,44 @@ async def delete_original_response(self) -> None:
...
+            The locale to use, this is handy if you want the
+            translation for a specific locale.
+            Defaults to the user's :attr:`.locale`.
+        data: Optional[Any]
+            The extraneous data that is being translated.
+            if not specified, either :attr:`.command`
+            or :attr:`.message` will be passed, depending on which is
+            available in the context.
+            This can be ``None``.
+        """
```
Comment (nerdguyahmad) [inline discord/interactions.py:530]:
```suggestion

        Returns
        -------
        Optional[:class:`str`]
            The translated string.
        """
```

File: discord/interactions.py [hunk 3]
```diff
@@ -503,6 +504,44 @@ async def delete_original_response(self) -> None:
             proxy_auth=http.proxy_auth,
         )
 
+    async def translate(
+        self, string: Union[str, locale_str], *, locale: Locale = MISSING, data: Optional[Any] = MISSING
+    ) -> Optional[Any]:
+        """|coro| Translates a string using the set :class:`~discord.app_commands.Translator`.
```
Comment (Gobot1234) [inline discord/interactions.py:510]:
```suggestion
        """|coro|
        Translates a string using the set :class:`~discord.app_commands.Translator`.
```

File: discord/interactions.py [hunk 4]
```diff
@@ -503,6 +504,44 @@ async def delete_original_response(self) -> None:
...
+            The string to translate.
+            :class:`~discord.app_commands.locale_str` can be used to add more context,
+            information, or any metadata necessary.
+        locale: :class:`Locale`
+            The locale to use, this is handy if you want the
+            translation for a specific locale.
+            Defaults to the user's :attr:`.locale`.
+        data: Optional[Any]
+            The extraneous data that is being translated.
+            if not specified, either :attr:`.command`
```
Comment (Gobot1234) [inline discord/interactions.py:526]:
```suggestion
            If not specified, either :attr:`.command`
```

File: discord/interactions.py [hunk 5]
```diff
@@ -503,6 +504,44 @@ async def delete_original_response(self) -> None:
...
+        ----------
+        string: Union[:class:`str`, :class:`~discord.app_commands.locale_str`]
+            The string to translate.
+            :class:`~discord.app_commands.locale_str` can be used to add more context,
+            information, or any metadata necessary.
+        locale: :class:`Locale`
+            The locale to use, this is handy if you want the
+            translation for a specific locale.
+            Defaults to the user's :attr:`.locale`.
+        data: Optional[Any]
```
Comment (nerdguyahmad) [inline discord/interactions.py:524]: The rest of documentation does not explicitly documents the type for parameters of `Any` type.
```suggestion
        data:
```

File: discord/interactions.py (hunk 1 above)
Comment (Bluesy1) [inline discord/interactions.py:509]:
```suggestion
    ) -> Optional[str]:
```
Translator.translate() returns `Optional[str]`, so shouldn't this also return that?

File: discord/interactions.py [hunk 6]
```diff
@@ -503,6 +504,44 @@ async def delete_original_response(self) -> None:
             proxy_auth=http.proxy_auth,
         )
 
+    async def translate(
+        self, string: Union[str, locale_str], *, locale: Locale = MISSING, data: Optional[Any] = MISSING
```
Comment (nerdguyahmad) [inline discord/interactions.py:508]: `Optional` is redundant here.
```suggestion
        self, string: Union[str, locale_str], *, locale: Locale = MISSING, data: Any = MISSING
```

**Review by nerdguyahmad — CHANGES_REQUESTED** [review]
Few typing changes

File: discord/interactions.py (hunk 3 above)
Comment (NCPlayz) [inline discord/interactions.py:510]:
```suggestion
        """|coro|
        
        Translates a string using the set :class:`~discord.app_commands.Translator`.
```

File: discord/interactions.py (hunk 6 above)
Comment (Gobot1234) [inline discord/interactions.py:508]:
```suggestion
        self, string: Union[str, locale_str], *, locale: Locale = MISSING, data: Any = MISSING
```
None is seemly never used as the type of data

File: discord/interactions.py [hunk 7]
```diff
@@ -503,6 +504,44 @@ async def delete_original_response(self) -> None:
...
+        locale: :class:`Locale`
+            The locale to use, this is handy if you want the
+            translation for a specific locale.
+            Defaults to the user's :attr:`.locale`.
+        data: Optional[Any]
+            The extraneous data that is being translated.
+            if not specified, either :attr:`.command`
+            or :attr:`.message` will be passed, depending on which is
+            available in the context.
+            This can be ``None``.
```
Comment (Gobot1234) [inline discord/interactions.py:529]: See first comment

**Review by Gobot1234 — COMMENTED** [review]
Line lengths in the docstring are also a bit inconsistent

**Review by nerdguyahmad — APPROVED** [review]

**Pull request merged**